[Instrumentation.Runtime] Release Runtime instrumentation version 1.0.0

xiang17 opened the pull request:
Release Runtime instrumentation version 1.0.0.

## Changes

Update `CHANGELOG.md`.

For significant contributions please make sure you have completed the following items:

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [x] Design discussion issue #335.

cijothomas commented:
Couple of things to check before stable release:

1. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0a7b14d2a7e319cc43ee321a2f705ff85b84c050/src/OpenTelemetry.Instrumentation.Runtime/README.md#troubleshooting dont see any actual logging? is this accurate?
2. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0a7b14d2a7e319cc43ee321a2f705ff85b84c050/src/OpenTelemetry.Instrumentation.Runtime/RuntimeInstrumentOptions.cs#L22 - The name sounds incorrect to me.. `RuntimeInstrumentationOptions` would be more apt.
3. There is not much public API, but need to make sure no breaking changes once 1.0.0 is shipped. Consider using public api analyzer before releasing stable version of any component.

utpilla commented:
Please enable CodeAnalysis as well.

xiang17 commented:
> Couple of things to check before stable release:
> 
>     1. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0a7b14d2a7e319cc43ee321a2f705ff85b84c050/src/OpenTelemetry.Instrumentation.Runtime/README.md#troubleshooting dont see any actual logging? is this accurate?
> 
>     2. https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/0a7b14d2a7e319cc43ee321a2f705ff85b84c050/src/OpenTelemetry.Instrumentation.Runtime/RuntimeInstrumentOptions.cs#L22
>         - The name sounds incorrect to me.. `RuntimeInstrumentationOptions` would be more apt.
> 
>     3. There is not much public API, but need to make sure no breaking changes once 1.0.0 is shipped. Consider using public api analyzer before releasing stable version of any component.

1. There is no logging. Removed the description from README.md: pending in #553.
2. Updated and merged.
3. Enabled PublicApiAnalyzer and merged the PR. Also updated this PR to move the PublicAPI from Unshipped to Shipped.
